lib/div-alg.gi

```diff
@@ -134,28 +134,54 @@ end);
 # and 2.    
 ################################
 InstallGlobalFunction( LocalIndexAtOddP, function(A,q)
-local m,F,n,a,b,c,n1,e,f,h,f1,e1,k;
+local m,F,a1,n,a,b,c,m1,m2,m3,g,g1,g2,g3,n1,e,f,h,f1,e1,k;
 
 m:=1; 
 F:=A[2];
+a1:=PrimitiveElement(F);
+m1:=A[3];
 a:=A[4][1];
 b:=A[4][2];
 c:=A[4][3];
-n:=Lcm(Conductor(F),A[3]);
+
+n:=Conductor(F);
+if IsOddInt(n) then n:=2*n; fi;
+for m2 in [1..n] do 
+  if E(n)^m2 in F then 
+     break; 
+  fi;
+od;  
+g:=Order((E(n)^m2)^a);
+
+g1:=E(m1);
+for m2 in [1..(a-1)] do 
+g1:=E(m1)*g1^b; 
+od; 
+g1:=Order(g1);
+
+g2:=Order(E(m1)^c);
+g3:=Lcm(g,g1);
+if (g3/g2 in Integers) then 
+ m:=1;
+else 
+
+n:=Lcm(Conductor(F),m1);   
 n1:=PDashPartOfN(n,q);
 ##########################
 e:=RamificationIndexAtP(F,n,q);
-if e>1 and c>0 and A[3]/q in PositiveIntegers then 
+if e>1 and c>0 and m1/q in PositiveIntegers then 
```

Comment:
Here and below one can use `IsPosInt` instead of `in` triggering method selection.

Reply:
There are more cases of using `in PositiveIntegers` - maybe those can be all updated in a separate commit.

Reply:
@drallenherman compare:
```
gap> for i in [1..10^8] do x:=IsPosInt(42);od;time;
4176
gap> for i in [1..10^8] do x:=42 in PositiveIntegers;od;time;
12833
```